src/main/frontend/common/RestClient.tsx

```diff
@@ -83,3 +80,27 @@ export async function getConsoleTextOffset(
     return null;
   }
 }
+
+const baseUrl = (): string =>
+  document.getElementsByTagName("head")[0].getAttribute("data-rooturl") ?? "";
```

Comment:
not really sure why this is a function either, but this is simpler:
```suggestion
  document.head.dataset.rooturl;
```

Reply:
I think some tests were failing when it wasn't a function due to `document` being called when it wasn't set by the framework, but will double check and see if I can fix

Reply:
ahh it is due to how things are loaded in the tests, I'll just making it a local variable within the `getResourceBundle` function